filebeat/inputsource/tcp/client.go

```diff
@@ -43,29 +43,37 @@ type splitHandler struct {
 }
 
 // ClientFactory returns a ConnectionHandler func
-type ClientFactory func(config Config) ConnectionHandler
+type ClientFactory func(config Config, conn net.Conn) ConnectionHandler
```

Comment:
Don't know why, but my OCD tells me `conn` should be the first parameter :)

Reply:
I don't have strong feeling but yes I can make 1st param :)

Reply:
Well I wont do this change, I've used the closeref strategy which remove the need for that.